### PR TITLE
Add rustc lint, warning when iterating over hashmaps

### PR DESCRIPTION
**Pull request by lcnr**
r? rust-lang/wg-incr-comp

### CODE REVIEW

File: src/test/ui-fulldeps/internal-lints/query_stability.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    for _ in x.iter() {}
+    //~^ ERROR using `iter`
+
+    for _ in Some(&mut x).unwrap().iter_mut() {}
+    //~^ ERROR using `iter_mut`
+
+    for _ in x {}
+    //~^ ERROR using `into_iter`
+}
```
Comment (estebank) [inline src/test/ui-fulldeps/internal-lints/query_stability.rs:24]: `fmt` is gonna complain about the missing newline

**Comment by estebank** [comment]
@bors r+

We don't need to mark the `Fx` versions?

**Comment by lcnr** [comment]
@bors r- 

somehow CI doesn't run at all for this PR and this lint causes a lot of warnings which is why i've opened https://github.com/rust-lang/compiler-team/issues/465 for this.

> We don't need to mark the `Fx versions?

The `Fx` versions are only type aliases, which are pretty much ignored by rustc (as you probably also painfully aware)

**Comment by petrochenkov** [comment]
@lcnr 
We have some lint naming conventions - https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#lints.
As you can see `allow(query_stability)` doesn't make much sense as a sentence.

**Review by estebank — APPROVED** [review]

**Pull request merged**